cedar-policy-core/src/ast/value.rs

```diff
@@ -484,38 +485,49 @@ impl std::fmt::Display for ValueKind {
                 fast,
                 authoritative,
             }) => {
-                match authoritative.len() {
-                    0 => write!(f, "[]"),
-                    n @ 1..=5 => {
-                        write!(f, "[")?;
-                        if let Some(rc) = fast {
-                            // sort the elements, because we want the Display output to be
-                            // deterministic, particularly for tests which check equality
-                            // of error messages
-                            for (i, item) in rc.as_ref().iter().sorted_unstable().enumerate() {
-                                write!(f, "{item}")?;
-                                if i < n - 1 {
-                                    write!(f, ", ")?;
-                                }
-                            }
-                        } else {
-                            // don't need to sort the elements in this case because BTreeSet iterates
-                            // in a deterministic order already
-                            for (i, item) in authoritative.as_ref().iter().enumerate() {
-                                write!(f, "{item}")?;
-                                if i < n - 1 {
-                                    write!(f, ", ")?;
-                                }
-                            }
+                write!(f, "[")?;
+                if let Some(rc) = fast {
+                    // sort the elements, because we want the Display output to be
+                    // deterministic, particularly for tests which check equality
+                    // of error messages
+                    for (i, item) in rc.as_ref().iter().sorted_unstable().enumerate() {
+                        write!(f, "{item}")?;
+                        if i < authoritative.len() - 1 {
+                            write!(f, ", ")?;
+                        }
+                    }
+                } else {
+                    // don't need to sort the elements in this case because BTreeSet iterates
+                    // in a deterministic order already
+                    for (i, item) in authoritative.as_ref().iter().enumerate() {
+                        write!(f, "{item}")?;
+                        if i < authoritative.len() - 1 {
+                            write!(f, ", ")?;
                         }
-                        write!(f, "]")?;
-                        Ok(())
                     }
-                    n => write!(f, "<set with {} elements>", n),
```

Comment:
Should we keep this case for the same reason we applied #887? Similarly should consider adding a case like this for the record branch.

Reply:
A goal for the policy pretty-printer is to produce valid, parseable syntax.  Is a similar goal reasonable here, that the value pretty-printer should produce syntax that is parseable as a Cedar expression?

Reply:
Fair point. 

But, for request validation errors in particular, I don't think it's a reasonable goal. It isn't ideal to have `InvalidContextError` send back the entire context when the context is invalid.

Reply:
Updated the request-validation error message to truncate after 5 key-value pairs in the `Context` record.  Left the `Display` impl for `Value` as printing valid Cedar syntax.  Is that a good compromise between the competing goals here?

Reply:
Refactored to new `bounded_display()` method on `Value` and `ValueKind`, which shares code with their `Display` impls.  Addresses bounding the display of large sets and records, including recursively; doesn't address large literals or extension values (?)

Reply:
Seems good enough for now. To me it feels more likely that the context record will be very large in the normal operation of an application than very large individual strings. If this turns about to be wrong we can truncate these as well.